tests/test_codecs.py

```diff
@@ -20,7 +24,8 @@ def check_request(req1, req2):
 
     enc = encoder(Request, send_body=True)
     dec = decoder(Request, Response)
-    req = Request(url="http://www.yandex.ru",method=b'GET', meta={b"test": b"shmest"}, headers={b'reqhdr': b'value'})
+    req = Request(url="http://www.yandex.ru", method=b'GET',
```

Comment:
I suggest to explicitly check the type for string objects in meta, to prove that your update codecs code really respects the type and preserves it.